engine/hatchery/swarm/swarm_util_kill.go

```diff
@@ -37,11 +38,32 @@ func (h *HatcherySwarm) killAndRemove(dockerClient *dockerClient, ID string) err
 				log.Error("hatchery> swarm> killAndRemove> unable to get model from registering container %s", container.Name)
 			} else {
 				if err := hatchery.CheckWorkerModelRegister(h, modelID); err != nil {
+					ctx, cancel := context.WithTimeout(context.Background(), time.Minute*2)
+					defer cancel()
+					logsOpts := types.ContainerLogsOptions{
+						Details:    true,
+						ShowStderr: true,
+						ShowStdout: true,
+						Timestamps: true,
+						Since:      "10s",
+					}
+					logsReader, errL := dockerClient.ContainerLogs(ctx, container.ID, logsOpts)
+					if errL != nil {
+						log.Error("hatchery> swarm> killAndRemove> cannot get logs from docker for containers service %s %v : %v", container.ID, container.Name, errL)
+					}
+
+					logs, errR := ioutil.ReadAll(logsReader)
+					defer logsReader.Close()
+					if errR != nil {
+						log.Error("hatchery> swarm> killAndRemove> cannot read logs for containers service %s %v : %v", container.ID, container.Name, errR)
+					}
+
 					var spawnErr = sdk.SpawnErrorForm{
 						Error: err.Error(),
+						Logs:  logs,
 					}
 					if err := h.CDSClient().WorkerModelSpawnError(modelID, spawnErr); err != nil {
-						log.Error("CheckWorkerModelRegister> error on call client.WorkerModelSpawnError on worker model %d for register: %s", modelID, err)
+						log.Error("hatchery> swarm> killAndRemove> error on call client.WorkerModelSpawnError on worker model %d for register: %s", modelID, err)
```

Comment:
Can use log.Error("%v", sdk.WrapError(err, "error on call client.WorkerModelSpawnError on worker model %d for register", modelID)) this will print the call stack auto.

Reply:
in this case the callstack is useless